### PR TITLE
feat(CD): Add Continuous Deployment pipeline to GCP 

### PR DESCRIPTION
**Pull request by bredamatt**
## Description

- Added a new project in the arweave.org in GCP called `hyperbeam-cd`.
- Created a new IAM user for the service account with the privileges to manage compute in the new GCP project.
- Created the credentials for the service account and used these for authenticating to GCP from the GitHub action runner
- Let the new service account assume the compute editor service account in the new project
- Separated the GitHub action into build, deploy, test, and cleanup jobs

### Credentials

The credentials are stored as a GitHub secret named `CD_SERVICE_ACCOUNT` containing the GCP service account credentials used for authenticating the runner to GCP. 

The service account was created as follows after authenticating to the `hyperbeam-cd` project using `gcloud`:

```sh
$ gcloud iam service-accounts create hyperbeam-cd-gha \
    --description="Service account for the hyperbeam-cd project" \
    --display-name="hyperbeam-cd-gha"
```

the `.json` credentials file was subsequently created as follows:
```sh
$ gcloud iam service-accounts keys create "hyperbeam-cd-gha.json" \
 --iam-account "hyperbeam-cd-gha@hyperbeam-cd.iam.gserviceaccount.com"
```

I then made sure to store the `.json` content in a GitHub secret associated with this repository.

The workflow uses the GitHub secret to authenticate to GCP using the `google-github-actions/auth` action thereafter, generating a token for authNZ purposes.

It should also be noted that adding the service account credentials to GitHub is not enough from a permissions perspective. The service account must also have the right permissions in GCP.
I used the GCP console to make sure the right permissions were attached to the service account. 
First, I granted the service account the Compute Admin role for the service account in the new project.
Then, I made sure the default Compute editor IAM user in the project can be used by the service account.

### Image tag

Note that the Packer / GCP image is built with a certain tag consisting of the last 7 chars of the latest commit-hash in addition to a timestamp. This means we have a distinctive way of separating one version from another.

## Attached documentation in the PR 

For further docs, see the `cd.md` file. The directory of this file should be used for further documenting additional workflows.

### CODE REVIEW

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -0,0 +1,138 @@
+name: Build and Deploy AO/HyperBEAM
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  GCP_IMAGE_NAME: hyperbeam-image
+  GCP_PROJECT: hyperbeam-cd
```
Comment (bredamatt) [inline .github/workflows/cd.yaml:10]: This project was made specifically for this branch. It is assumed that the GCP project with this name already exists, hence if a new project needs to be used, then that needs to be created prior to changing this line.

File: rebar.config [hunk 2]
```diff
@@ -39,7 +39,9 @@
 {deps, [
     {b64fast, {git, "https://github.com/ArweaveTeam/b64fast.git", {ref, "58f0502e49bf73b29d95c6d02460d1fb8d2a5273"}}},
     {jiffy, {git, "https://github.com/ArweaveTeam/jiffy.git", {ref, "74c956defa9116c85d76f77c3e9b5bd6de7bd39a"}}},
-    {cowboy, {git, "https://github.com/ninenines/cowboy", {tag, "2.12.0"}}}
+    {cowboy, {git, "https://github.com/ninenines/cowboy", {tag, "2.12.0"}}},
+    {gun, "2.0.1"},
```
Comment (oltarasenko) [inline rebar.config:43]: To double check here, do we need `gun` and `gproc` here?

Comment (bredamatt) [reply rebar.config:43]: When building on my Mac I seemed to need it at least. It was not there before so I recon someone building it elsewhere maybe didn't need it. That being said I am new to Erlang so wouldn't know for certain.

**Review by oltarasenko — APPROVED** [review]
It would be nice to get the changes to rebar.config in.

**Comment by bredamatt** [comment]
> It would be nice to get the changes to rebar.config in.

Just merged main into this. Will merge this into main now.

**Pull request merged**